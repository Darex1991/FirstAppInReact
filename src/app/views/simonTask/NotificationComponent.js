import React, { Component, PropTypes } from 'react';
import { Checkbox } from 'react-form'

export default class NotificationComponent extends Component {
  static propTypes = {
    dataBase: PropTypes.array.isRequired
  };

  constructor(props){
    super(props);
  }

  render() {
    let fields = this.props.dataBase.phones.map( (item, i) => {
      return (
        <tr key={i} >
          <td><p>{item.name} <span>({item.value}) </span></p></td>
          <td><Checkbox field={item.name}/></td>
        </tr>
      )});

    return (
      <div className="profile default-input">
        <h2> Notification </h2>

        <div>
          <table>
            <tbody>
              <tr>
                <th> </th>
                <th>Standard Prior</th>
                <th>High Prior</th>
              </tr>
              <tr>
                <td><p> Email <span>(asdasd@asdasd.pl)</span></p></td>
                <td><Checkbox field='emailCheckbox'/></td>
                <td><Checkbox field='emailCheckboxPremium'/></td>
              </tr>
              {fields}
            </tbody>
          </table>
        </div>
      </div>
    )
  }
}
