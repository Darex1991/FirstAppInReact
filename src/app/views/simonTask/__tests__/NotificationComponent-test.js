import React from 'react';
import renderer from 'react-test-renderer';
import NotificationComponent from '../NotificationComponent';
test('Should render NotificationComponent correctly', () => {
    const tree = renderer.create(
        <NotificationComponent /> 
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
    