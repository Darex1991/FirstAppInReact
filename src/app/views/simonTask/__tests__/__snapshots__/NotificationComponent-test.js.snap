exports[`test Should render NotificationComponent correctly 1`] = `
<div
  className="profile default-input">
  <h2>
    Notification
  </h2>
  <div>
    <table>
      <tbody>
        <tr>
          <th>
             
          </th>
          <th>
            Standard Prior
          </th>
          <th>
            High Prior
          </th>
        </tr>
        <tr>
          <td>
            <p>
               Email 
              <span>
                (asdasd@asdasd.pl)
              </span>
            </p>
          </td>
          <td>
            <FormInputCheckbox
              field="emailCheckbox" />
          </td>
          <td>
            <FormInputCheckbox
              field="emailCheckboxPremium" />
          </td>
        </tr>
        <tr>
          <td>
            <p>
              asd
               
              <span>
                (
                val 1
                ) 
              </span>
            </p>
          </td>
          <td>
            <input
              checked={true}
              onChange={[Function]}
              type="checkbox" />
          </td>
          <td>
            <input
              checked={true}
              onChange={[Function]}
              type="checkbox" />
          </td>
        </tr>
        <tr>
          <td>
            <p>
              asdasdasd
               
              <span>
                (
                val 122222
                ) 
              </span>
            </p>
          </td>
          <td>
            <input
              checked={false}
              onChange={[Function]}
              type="checkbox" />
          </td>
          <td>
            <input
              checked={false}
              onChange={[Function]}
              type="checkbox" />
          </td>
        </tr>
        <tr>
          <td>
            <p>
              asdasdasd
               
              <span>
                (
                val asdasd
                ) 
              </span>
            </p>
          </td>
          <td>
            <input
              checked={false}
              onChange={[Function]}
              type="checkbox" />
          </td>
          <td>
            <input
              checked={true}
              onChange={[Function]}
              type="checkbox" />
          </td>
        </tr>
        <tr>
          <td>
            <p>
              222
               
              <span>
                (
                val213123 1
                ) 
              </span>
            </p>
          </td>
          <td>
            <input
              checked={true}
              onChange={[Function]}
              type="checkbox" />
          </td>
          <td>
            <input
              checked={false}
              onChange={[Function]}
              type="checkbox" />
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
